Cubical/Codata/M/AsLimit/Coalg/Properties.agda

```diff
@@ -0,0 +1,207 @@
+{-# OPTIONS --cubical --guardedness #-} --safe
+
+open import Cubical.Data.Unit
+open import Cubical.Data.Sigma
+open import Cubical.Data.Nat
+open import Cubical.Data.Prod
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function using ( _∘_ ; idfun )
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.GroupoidLaws
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Path
+
+open import Cubical.Functions.Embedding
+open import Cubical.Functions.FunExtEquiv
+
+open import Cubical.Codata.M.AsLimit.helper
+
+open import Cubical.Codata.M.AsLimit.Coalg.Base
+open import Cubical.Codata.M.AsLimit.Container
+open import Cubical.Codata.M.AsLimit.M
+
+module Cubical.Codata.M.AsLimit.Coalg.Properties where
+
+open Iso
+
+-- Construction of Coalgebras from M-types, defined in
+-- https://arxiv.org/pdf/1504.02949.pdf
+-- "Non-wellfounded trees in Homotopy Type Theory"
+-- Benedikt Ahrens, Paolo Capriotti, Régis Spadotti
+
+-----------------------------------------------------------------------------
+-- The limit of a Polynomial functor over a Container is a Final Coalgebra --
+-----------------------------------------------------------------------------
+
+Ps : ∀ {ℓ} -> (S : Container ℓ) -> (C,γ : Coalg₀ S) -> Container ℓ
+Ps S (C , γ) = C , λ x → P₀ S C
+
+Ms : ∀ {ℓ} -> (S : Container ℓ) -> Container ℓ
+Ms S = M S , λ x → P₀ S (M S)
+
+M-coalg : ∀ {ℓ} {S : Container ℓ} -> Coalg₀ S
+M-coalg {S = S} =
+  (M S) , out-fun
+
+PM-coalg : ∀ {ℓ} {S : Container ℓ} -> Coalg₀ S
+PM-coalg {S = S} = (P₀ S (M S)) , (P₁ out-fun)
+
+Final : ∀ {ℓ} {S : Container ℓ} -> Set (ℓ-suc ℓ)
+Final {S = S} = Σ[ X,ρ ∈ Coalg₀ S ] ∀ (C,γ : Coalg₀ S) -> isContr ((C,γ) ⇒ (X,ρ))
+
+--------------------------------------------------------
+-- Properties of Bisimulations and (Final) Coalgebras --
+--------------------------------------------------------
+
+U : ∀ {ℓ} {S : Container ℓ} (C,γ : Coalg₀ S) -> Set ℓ
```

Comment:
What does U stand for?

Reply:
I followed the notation in the paper, so I can (and probably should) change it to something less generic.